Source/Core/DiscIO/DirectoryBlob.cpp

```diff
@@ -896,11 +912,12 @@ DirectoryBlobPartition::DirectoryBlobPartition(
   if (sys_callback)
     sys_callback(&sys_nodes);
 
-  SetDiscHeader(ExtractNodeToVector(&sys_nodes, &m_disc_header));
-  SetDiscType(is_wii);
-  SetBI2(ExtractNodeToVector(&sys_nodes, &m_bi2));
+  disc_header = ExtractNodeToVector(&sys_nodes, &disc_header);
+  disc_header.resize(DISCHEADER_SIZE);
+  SetDiscType(is_wii, disc_header);
+  SetBI2(ExtractNodeToVector(&sys_nodes, bi2.data()));
   const u64 new_dol_address =
-      SetApploader(ExtractNodeToVector(&sys_nodes, &m_apploader), "apploader");
+      SetApploader(ExtractNodeToVector(&sys_nodes, apploader.data()), "apploader");
```

Comment:
```suggestion
      SetApploader(ExtractNodeToVector(&sys_nodes, &apploader), "apploader");
```

The pointer returned by `apploader.data()` can change when you change the contents of the vector, including when moving from the vector, so you can't use it as a stable identifier.

The same problem exists other times you're creating a `FSTBuilderNode`, including when doing it by calling `BuildSingleFileNode`.